lib/handler/configurator/proxy.c

```diff
@@ -421,6 +536,11 @@ void h2o_proxy_register_configurator(h2o_globalconf_t *conf)
                                     H2O_CONFIGURATOR_FLAG_PATH | H2O_CONFIGURATOR_FLAG_EXPECT_SCALAR |
                                         H2O_CONFIGURATOR_FLAG_EXPECT_SEQUENCE | H2O_CONFIGURATOR_FLAG_DEFERRED,
                                     on_config_reverse_url);
+    /* support selecting wanted load balancing strategy */
+    h2o_configurator_define_command(&c->super, "proxy.reverse.balancer",
+                                    H2O_CONFIGURATOR_FLAG_PATH | H2O_CONFIGURATOR_FLAG_EXPECT_SCALAR |
+                                    H2O_CONFIGURATOR_FLAG_EXPECT_MAPPING,
+                                    on_config_reverse_balancer);
```

Comment:
How about letting the user specify the balancing logic as an argument to `proxy.reverse.url`? I think that would be more straight-forward. For example, something like:
```
proxy.reverse.url:
  backends:
    - url: http://origin1.example.com:8080/
      weight: 10
    - url: http://origin2.example.com:8080/
      weight: 20
  load-balance: least-conn
```